cpg_utils/cloud.py

```diff
@@ -322,3 +323,42 @@ def is_member_in_cached_group(
     return member.lower() in get_cached_group_members(
         group, members_cache_location=members_cache_location
     )
+
+
+def get_path_components_from_gcp_path(path: str) -> dict[str, str] | None:
+    """
+    Return the {bucket_name}, {dataset}, {bucket_type}, {subdir}, and {file} for GS only paths
+    Uses regex to match the full bucket name, dataset name, bucket type (e.g. 'test', 'main-upload', 'release'),
+    subdirectory, and the file name.
+    """
+
+    bucket_types = ['archive', 'hail', 'main', 'test', 'release']
+
+    rmatch_str = (
+        r'gs://(?P<bucket>cpg-(?P<dataset>[\w-]+)-(?P<bucket_type>['
+        + '|'.join(s for s in bucket_types)
+        + r']+[-\w]*))/(?P<suffix>.+/)?(?P<file>.*)$'
+    )
```

Comment:
this is cool, but unless it's parametrised from config you may as well just make the fixed string here?

Reply:
Not really sure what you mean by this, are you saying to scrap the bucket_types list and add them into the string?

Reply:
Pretty much. I actually like keeping this in, as it shows the syntax if we want to entertain a wider range of bucket categories later.